testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -1218,3 +1231,112 @@ def test_record_type_sel():
 
     assert len(u.select_atoms('name CA and not record_type HETATM')) == 30
     assert len(u.select_atoms('name CA and record_type HETATM')) == 2
+
+
+def test_element_sel():
+    # test auto-topattr addition of string
+    u = mda.Universe(PDB_elements)
+    assert len(u.select_atoms("element H")) == 8
+    assert len(u.select_atoms("same element as index 12")) == 8
+
+
+def test_chain_sel():
+    u = mda.Universe(PDB_elements)
+    assert len(u.select_atoms("chainID A")) == len(u.atoms)
+
+
+@pytest.fixture()
+def u_fake_masses():
+    u = mda.Universe(TPR)
+    u.atoms[-10:].masses = - (np.arange(10) + 1.001)
+    u.atoms[:5].masses = 0.1 * 3  # 0.30000000000000004
+    u.atoms[5:10].masses = 0.30000000000000001
+    return u
+
+
+@pytest.mark.parametrize("selstr,n_atoms, selkwargs", [
+    ("mass 0.8 to 1.2", 23844, {}),
+    ("mass 8e-1 to 1200e-3", 23844, {}),
+    ("mass -5--3", 2, {}),  # select -5 to -3
+    ("mass -5 --3", 2, {}),  # spacing
+    ("mass -5- -3", 2, {}),  # spacing
+    ("mass -3 : -5", 0, {}),  # wrong way around
+    # float equality
+    ("mass 0.3", 5, {"rtol": 0, "atol": 0}),
+    ("mass 0.3", 5, {"rtol": 1e-22, "atol": 1e-22}),
+    # 0.30000000000000001 == 0.3
+    ("mass 0.3 - 0.30000000000000004", 10, {}),
+    ("mass 0.30000000000000004", 5, {"rtol": 0, "atol": 0}),
+    ("mass 0.3 0.30000000000000001", 5, {"rtol": 0, "atol": 0}),
+    # float near-equality
+    ("mass 0.3", 10, {}),
+    ("mass 0.30000000000000004", 10, {}),
+    ("mass 0.3 0.30000000000000001", 10, {}),
+    # prop thingy
+    ("prop mass == 0.3", 10, {}),
+    ("prop mass == 0.30000000000000004", 10, {}),
+    ("prop mass == 0.30000000000000004", 5, {"rtol": 0, "atol": 0}),
+])
+def test_mass_sel(u_fake_masses, selstr, n_atoms, selkwargs):
+    # test auto-topattr addition of float (FloatRangeSelection)
+    ag = u_fake_masses.select_atoms(selstr, **selkwargs)
+    assert len(ag) == n_atoms
+
+def test_mass_sel_warning(u_fake_masses):
+    warn_msg = (r"Using float equality .* is not recommended .* "
+                r"we recommend using a range .*"
+                r"'mass -0.6 to 1.4'.*"
+                r"use the `atol` and `rtol` keywords")
+    with pytest.warns(SelectionWarning, match=warn_msg):
+        u_fake_masses.select_atoms("mass 0.4")
+
+
+@pytest.mark.parametrize("selstr,n_res", [
+    ("resnum -10 to 3", 14),
+    ("resnum -5--3", 3),  # select -5 to -3
```

Comment:
can we include `-10:3` `-5 - -3` etc to make sure the regex is correct here

Reply:
Done! Thank you :-)